modules/gapi/src/backends/common/serialization.hpp

```diff
@@ -0,0 +1,306 @@
+#ifndef OPENCV_GAPI_COMMON_SERIALIZATION_HPP
+#define OPENCV_GAPI_COMMON_SERIALIZATION_HPP
+
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+//
+// Copyright (C) 2020 Intel Corporation
+
+#include <iostream>
+#include <fstream>
+#include <string.h>
+
+#include <ade/util/iota_range.hpp> // used in the vector<</>>
+
+#include "compiler/gmodel.hpp"
+
+namespace cv {
+namespace gimpl {
+namespace s11n {
+
+struct GSerialized {
+    std::vector<cv::gimpl::Op> m_ops;
+    std::vector<cv::gimpl::Data> m_datas;
+    cv::gimpl::DataObjectCounter m_counter;
+    cv::gimpl::Protocol m_proto;
+};
+
+////////////////////////////////////////////////////////////////////////////////
+// Stream interfaces, so far temporary
+namespace I {
+    struct GAPI_EXPORTS OStream {
+        virtual ~OStream() = default;
+
+        // Define the native support for basic C++ types at the API level:
+        virtual OStream& operator<< (bool) = 0;
+        virtual OStream& operator<< (char) = 0;
+        virtual OStream& operator<< (unsigned char) = 0;
+        virtual OStream& operator<< (short) = 0;
+        virtual OStream& operator<< (unsigned short) = 0;
+        virtual OStream& operator<< (int) = 0;
+        virtual OStream& operator<< (std::size_t) = 0;
```

Comment:
As  [size_t](https://en.cppreference.com/w/cpp/types/size_t) is an alias the clash with other overloads is inevitable. IMHO more robust solution would be : 
1.  Add overloads for all possible `unsigned` integer types `size_t` might be aliased to
2. Add a `static_assert` next to overload declarations that `size_t` is actually one of those types (to be sure that exact match for `size_t` is present in set of overloaded functions)  
  
Then, in contexts where `size_t` is serialized one of those overloads will be chosen (And no type conversion will happen, as there will be an exact match).

Reply:
let just have an uint32_t version and cast size_t at the place where we pass it down.